### PR TITLE
feat: profile picture shown in lobby

### LINKED ISSUE #420: Show profile picture on lobby screen

### PR DESCRIPTION
**Pull request by jameshallam93**
closes #420 

Show profile picture in lobby. For empty spots, show no_player_img

### CODE REVIEW

File: packages/app/src/lobby/lobbyPage.tsx [hunk 1]
```diff
@@ -35,9 +35,9 @@ const LobbyPage: React.FunctionComponent = () => {
 
 	for (let i = players.length; i < MAX_PLAYERS_IN_GAME; i++) {
 		botElements.push(
-			<div key={i} className="player bot">
-				<span className="name">empty slot</span>
-			</div>
+			<LobbyPlayerBanner
```
Comment (Jameskmonger) [inline packages/app/src/lobby/lobbyPage.tsx:38]: This also needs a key! Even if you're not using `map`, you need a key if you're rendering things inside an array (which we are, as `botElements` is an array)

File: packages/ui/src/lobbyPlayerBanner/lobbyPlayerBanner.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+						alt="no player image"
+					/>
+					<div className="spacer" />
+					empty slot
+				</span>
+			</div>
+		);
+	}
+	return (
+		<div key={player?.id} className="player">
```
Comment (Jameskmonger) [inline packages/ui/src/lobbyPlayerBanner/lobbyPlayerBanner.tsx:26]: Move the key up a level, individual components should not know about their own key, the key is the concern of the thing rendering multiple of them

File: packages/ui/stories/lobbyPlayerBanner/lobbyPlayerBanner.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+	player: createLobbyPlayer(null, false)
+}
+export const Developer = Template.bind({});
+Developer.args = {
+	player: createLobbyPlayer(PlayerTitle.Developer, false)
+}
+export const BotPlayer = Template.bind({});
+BotPlayer.args = {
+	player: createLobbyPlayer(null, true)
+}
```
Comment (Jameskmonger) [inline packages/ui/stories/lobbyPlayerBanner/lobbyPlayerBanner.stories.tsx:41]: Love it ,nice work and hope you liked using storybook

File: packages/ui/src/lobbyPlayerBanner/lobbyPlayerBanner.tsx [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+
+type Props = {
+	player: LobbyPlayer | null;
+};
+
+const LobbyPlayerBanner: React.FunctionComponent<Props> = ({ player }) => {
+	if (!player) {
+		return (
+			<div key={Math.floor(Math.random() * 1000)} className="player bot">
```
Comment (Jameskmonger) [inline packages/ui/src/lobbyPlayerBanner/lobbyPlayerBanner.tsx:13]: avoid random keys https://codeburst.io/how-to-not-react-common-anti-patterns-and-gotchas-in-react-40141fe0dcd

**Review by Jameskmonger — CHANGES_REQUESTED** [review]

**Review by jameshallam93 — COMMENTED** [review]
Accidental review

**Review by Jameskmonger — APPROVED** [review]

**Pull request merged**